src/js/pageAction.js

```diff
@@ -34,4 +34,20 @@ async function init() {
   MozillaVPN.handleContainerList(identities);
 }
 
+async function applyTheme() {
+  const { currentTheme } = await browser.storage.local.get("currentTheme");
+  const popup = document.getElementsByTagName("html")[0];
+
+  if (typeof currentTheme === "undefined" || currentTheme === "auto") {
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+      popup.setAttribute("data-theme", "dark");
+    } else {
+      popup.setAttribute("data-theme", "light");
+    }
+  } else {
+    popup.setAttribute("data-theme", currentTheme);
+  }
+}
+
+applyTheme();
```

Comment:
Nested `if/else` blocks smell like too much logic in a single function. Can we refactor into a helper function here?

```suggestion
function getTheme(currentTheme, window) {
  if (typeof currentTheme !== "undefined" && currentTheme !== "auto") {
    return currentTheme;
  }
  if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
    return "dark";
  }
  return "light";
}

async function applyTheme() {
  const { currentTheme } = await browser.storage.local.get("currentTheme");
  const popup = document.getElementsByTagName("html")[0];
  const theme = getTheme(currentTheme, window);
  popup.setAttribute("data-theme", theme);
}

applyTheme();
```

Reply:
Fixed via [a29285f](https://github.com/mozilla/multi-account-containers/pull/1964/commits/a29285f0fa730dd27e406cc17decbbac60213f4a)